### ISSUE TITLE
P-value meaning

### ISSUE THREAD

**Issue by MoMaz123** [bhklab/survcomp#12]
Hi 
I have used your PKG and run below function for my test and train data. If the P-value  is sig what this means? 

Also can I use concordance.index function to check the difference between two seprate cox models (fit 1 and  fit2)?
 
Many  thanks indeed!

cox_model = coxph(Surv(train$ihd_time,train$ihd_ep) ~ ., data=train)
 
# Create survival estimates on validation data
pred_validation = predict (cox_model, newdata = test)
 
# Determine concordance
cindex_validation = concordance.index (pred_validation, surv.time = test$ihd_time,
                                       surv.event=test$ihd_ep, method = "noether",na.rm = T)
 
cindex_validation$c.index [1] 
0.6154352 
cindex_validation$lower [1] 
0.5851906  
cindex_validation$upper [1] 
0.6456798 
cindex_validation$p.value [1] 
7.396189e-14

**Comment by ChristopherEeles** [comment]
Hi @MoMaz123,

I assume you are asking about the p-value returned from the `concordance.index` function, in which case this is a test statistic to see if the concordance index between your predicted and true labels is statistically significantly different from 0.5 (_i.e._, from random). This explanation is available in the documentation for the function: `?concordance.index`. If you are asking about the p-value from `coxph`, I would direct you to the [survival](https://cran.r-project.org/web/packages/survival/index.html) package on CRAN since this model is not part of our package.

RE: comparing the results of two different model fits, assuming they are for the same set of samples (patients) I would recommend computing the concordance index for each model independently then comparing the magnitude and significance of the concordance index for each model. Assuming the models pass your selected significance cut-off, then the model with a higher concordance index would be the "better" model. 

If you are comparing the performance of the same model on two different sets of testing data then one would hope to see significant result in both cases with the value of the concordance index trending in the same direction. If you do not see this it could be a sign that your Cox model has overfit the training data and thus doesn't generalize well to new sample (patients).

Best,
Christopher Eeles
Software Developer
Haibe-Kains Lab | PM-Research
University Health Network

**Comment by MoMaz123** [comment]
Many  thanks for  this, 
I am trying to see if my C-stat is robust, can I produce a cross-validated/bootstrap C-stat in R (cox regression).
Thanks, mo

**Comment by ChristopherEeles** [comment]
No problem.

Reviewing my comments I also want to clarify interpretation of the concordance index, when comparing models we actually want `abs(concordance_index - 0.5)` to be maximized, since 0.5 is random. In the case of a concordance index below 0.5, that indicates negative concordance between ranks, we can just invert the ranking and we have a model doing better than random again.

Best,
Chris
